spanner/batch.go

```diff
@@ -291,10 +332,27 @@ func (t *BatchReadOnlyTransaction) Execute(ctx context.Context, p *Partition) *R
 				RequestOptions: p.rreq.RequestOptions,
 				ResumeToken:    resumeToken,
 			})
+			if client != nil {
```

Comment:
I think that the following is more idiomatic, and it also reduces nesting:

```golang
if err != nil` {
    return client, err
}
```

Reply:
You mean I check if err==nil then only go into the GFE Latency block?

Reply:
No, the idiomatic thing to do in Go is to return early if an error is returned by a function that is called. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Also, normally only one of `client` and `err` should be non-nil at this point. An idiomatic Go function should normally not return both a 'normal' return value and an error. So that means that if `err != nil` is true, then `client == nil` should also be true.